### PR TITLE
first commit here, added dto to all models , added controllers to all…

### PR DESCRIPTION
**Pull request by maks45**
… models

### CODE REVIEW

File: src/main/java/com/durov/maks/cinema/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        this.userService = userService;
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
+        return shoppingCartMapper.getShoppingCartResponseDto(shoppingCartService
+                .getByUser(userService.getUserById(userId)));
+    }
+
+    @PostMapping("/add-movie-session")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/durov/maks/cinema/controller/ShoppingCartController.java]: we should not create a new resource for post method
```suggestion
    @PostMapping
```

File: src/main/java/com/durov/maks/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 2]
```diff
@@ -52,4 +54,18 @@ public CinemaHall add(CinemaHall cinemaHall) {
...
+
+    @Override
+    public CinemaHall getById(Long cinemaHallId) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<CinemaHall> criteriaQuery = criteriaBuilder.createQuery(CinemaHall.class);
+            Root<CinemaHall> cinemaHallRoot = criteriaQuery.from(CinemaHall.class);
+            criteriaQuery.select(cinemaHallRoot).where(criteriaBuilder
+                    .equal(cinemaHallRoot.get("id"), cinemaHallId));
+            return session.createQuery(criteriaQuery).getSingleResult();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/durov/maks/cinema/dao/impl/CinemaHallDaoImpl.java]: return session.get(CinemaHall.class, cinemaHallId);
try this one(here and everywhere where you are getting entity by id)

File: src/main/java/com/durov/maks/cinema/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -50,4 +52,18 @@ public Movie add(Movie movie) {
...
+    @Override
+    public Movie getById(Long movieId) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Movie> criteriaQuery = criteriaBuilder.createQuery(Movie.class);
+            Root<Movie> movieRoot = criteriaQuery.from(Movie.class);
+            criteriaQuery.select(movieRoot).where(criteriaBuilder
+                    .equal(movieRoot.get("id"), movieId));
+            return session.createQuery(criteriaQuery).getSingleResult();
+        } catch (HibernateException e) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/durov/maks/cinema/dao/impl/MovieDaoImpl.java]: same

File: src/main/java/com/durov/maks/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -58,4 +58,20 @@ public MovieSession add(MovieSession movieSession) {
...
+    @Override
+    public MovieSession getMovieSessionById(Long movieSessionId) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<MovieSession> criteriaQuery = criteriaBuilder
+                    .createQuery(MovieSession.class);
+            Root<MovieSession> movieSessionRoot = criteriaQuery.from(MovieSession.class);
+            criteriaQuery.select(movieSessionRoot).where(criteriaBuilder
+                    .equal(movieSessionRoot.get("id"), movieSessionId));
+            return session.createQuery(criteriaQuery).getSingleResult();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/durov/maks/cinema/dao/impl/MovieSessionDaoImpl.java]: same

File: src/main/java/com/durov/maks/cinema/dao/impl/UserDaoImpl.java [hunk 5]
```diff
@@ -56,4 +56,18 @@ public User add(User user) {
...
+    @Override
+    public User getById(Long userId) {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<User> criteriaQuery = criteriaBuilder.createQuery(User.class);
+            Root<User> userRoot = criteriaQuery.from(User.class);
+            criteriaQuery.select(userRoot).where(criteriaBuilder
+                    .equal(userRoot.get("id"), userId));
+            return session.createQuery(criteriaQuery).getSingleResult();
+        } catch (HibernateException e) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/durov/maks/cinema/dao/impl/UserDaoImpl.java]: )

File: src/main/java/com/durov/maks/cinema/model/CinemaHall.java [hunk 6]
```diff
@@ -7,7 +7,7 @@
 import javax.persistence.Table;
 
 @Entity
-@Table(name = "cinema_hall")
+@Table(name = "cinema_halls")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/durov/maks/cinema/model/CinemaHall.java:10]:
```suggestion
@Table(name = "cinema-halls")
```
image

File: src/main/java/com/durov/maks/cinema/model/dto/moviesession/MovieSessionRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package com.durov.maks.cinema.model.dto.moviesession;
+
+import java.time.LocalDateTime;
+
+public class MovieSessionRequestDto {
+    private LocalDateTime date;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/durov/maks/cinema/model/dto/moviesession/MovieSessionRequestDto.java:6]: Could you please send a screenshot of the view of your JSON object which is corresponding to this DTO. Just curious.

File: src/main/java/com/durov/maks/cinema/controller/ShoppingCartController.java (hunk 1 above)
Comment (maks45) [reply src/main/java/com/durov/maks/cinema/controller/ShoppingCartController.java]: got it

File: src/main/java/com/durov/maks/cinema/model/CinemaHall.java (hunk 6 above)
Comment (maks45) [reply src/main/java/com/durov/maks/cinema/model/CinemaHall.java:10]: got-it)

File: src/main/java/com/durov/maks/cinema/dao/impl/CinemaHallDaoImpl.java (hunk 2 above)
Comment (maks45) [reply src/main/java/com/durov/maks/cinema/dao/impl/CinemaHallDaoImpl.java]: got it

File: src/main/java/com/durov/maks/cinema/model/dto/moviesession/MovieSessionRequestDto.java (hunk 7 above)
Comment (maks45) [reply src/main/java/com/durov/maks/cinema/model/dto/moviesession/MovieSessionRequestDto.java:6]: of course
json

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**